packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1089 @@
+  outlineWidth: 2,
+  outlineColor: 'focus-ring',
+  borderRadius: size(6)
+});
+
+function CellFocusRing(props: {isFocusVisible: boolean}) {
+  let {isFocusVisible} = props;
+  return <div role="presentation" className={cellFocus({isFocusVisible})} />;
+}
+
+const columnStyles = style({
+  height: '[inherit]',
+  boxSizing: 'border-box',
+  color: {
+    default: 'gray-800', // neutral-content-color-default
+    isHovered: 'gray-900', // neutral-content-color-hover
+    isPressed: 'gray-900', // neutral-content-color-down
+    isFocusVisible: 'gray-900' // neutral-content-color-key-focus
+  },
+  paddingX: {
+    default: 16,
+    isColumnResizable: 0
+  },
+  textAlign: {
+    align: {
+      start: 'start',
+      center: 'center',
+      end: 'end'
+    }
+  },
+  outlineStyle: 'none',
+  position: 'relative',
+  fontSize: 'control',
+  fontFamily: 'sans',
+  fontWeight: 'bold',
+  display: 'flex',
+  borderColor: 'gray-300',
+  borderTopWidth: 0,
+  borderBottomWidth: 1,
+  borderStartWidth: 0,
+  borderEndWidth: {
+    default: 0,
+    isColumnResizable: 1
+  },
+  borderStyle: 'solid'
+});
+
+export interface ColumnProps extends RACColumnProps {
+  /** Whether the column should render a divider between it and the next column. */
+  showDivider?: boolean,
+  /** Whether the column allows resizing. */
+  allowsResizing?: boolean,
+  /**
+   * The alignment of the column's contents relative to its allotted width.
+   * @default 'start'
+   */
+  align?: 'start' | 'center' | 'end',
+  /** The content to render as the column header. */
+  children?: ReactNode
 }
 
 export function Column(props: ColumnProps) {
+  let {isQuiet} = useContext(InternalTableContext);
+  let {isHeaderRowHovered} = useContext(InternalTableHeaderContext);
+  let {allowsResizing, children, align = 'start'} = props;
+  let isColumnResizable = allowsResizing;
+
   return (
-    <AriaColumn {...props}>
-      {({allowsSorting, sortDirection}) => (
+    <RACColumn {...props} style={{borderInlineEndColor: 'transparent'}} className={renderProps => columnStyles({...renderProps, isQuiet, isColumnResizable, align})}>
+      {({allowsSorting, sortDirection, isFocusVisible, sort, startResize, isHovered}) => (
         <>
-          {props.children}
-          {allowsSorting && (
-            <span aria-hidden="true" className="sort-indicator">
-              {sortDirection === 'ascending' ? '▲' : '▼'}
-            </span>
-          )}
+          {/* Note this is mainly for column's without a dropdown menu. If there is a dropdown menu, the button is styled to have a focus ring for simplicity
+          (no need to juggle showing this focus ring if focus is on the menu button and not if it is on the resizer) */}
+          {/* Separate absolutely positioned element because appyling the ring on the column directly via outline means the ring's required borderRadius will cause the bottom gray border to curve as well */}
+          <CellFocusRing isFocusVisible={isFocusVisible} />
```

Comment:
I know we were trying to cut down on all the extra elements but relying on the outline styles for the column directly for the focus ring ran into the above issues

Reply:
probably could only mount this element when focus is actually visible?
```suggestion
          {/* Note this is mainly for column's without a dropdown menu. If there is a dropdown menu, the button is styled to have a focus ring for simplicity
          (no need to juggle showing this focus ring if focus is on the menu button and not if it is on the resizer) */}
          {/* Separate absolutely positioned element because appyling the ring on the column directly via outline means the ring's required borderRadius will cause the bottom gray border to curve as well */}
          {isFocusVisible && <CellFocusRing />}
```